File: 1287-element-appearing-more-than-25-in-sorted-array/1287-element-appearing-more-than-25-in-sorted-array.go
func findSpecialInteger(arr []int) int {
    targetCount := len(arr) / 4
    
    count := 0
    curr := arr[0]
    for _, num := range arr {
        if curr != num {
            curr = num
            count = 1
            continue
        }
        
        count++
        
        if count > targetCount {
            break
        }
    }
    return curr
}

File: 1287-element-appearing-more-than-25-in-sorted-array/1287-element-appearing-more-than-25-in-sorted-array.py
class Solution:
    def findSpecialInteger(self, arr: List[int]) -> int:
        targetCount = len(arr) / 4
        
        count = 0
        curr = None
        
        for num in arr:
            if num != curr:
                count = 1
                curr = num
                continue
            
            count += 1
            
            if count > targetCount:
                break
                
        return curr
                